test/integration/fleetscope/fleetscope_test.go

```diff
@@ -163,6 +169,27 @@ func TestFleetscope(t *testing.T) {
 					assert.Equal(gkeScopes, opGKEScopes.Get("name").String(), fmt.Sprintf("The GKE Namespace should be %s", gkeScopes))
 					assert.True(opGKEScopes.Exists(), "Namespace %s should exist", gkeScopes)
 				}
+				gkeMeshCommand := fmt.Sprintf("beta container fleet mesh describe --project %s --format='json(membershipStates)'", clusterProjectID)
+				pollMeshProvisioning := func(cmd string) func() (bool, error) {
+					return func() (bool, error) {
+						retry := false
+						result := gcloud.Runf(t, cmd)
+						if len(result.Array()) < 1 {
+							return true, nil
+						}
+						for _, memberShipName := range membershipNamesProjectNumber {
+							dataPlaneManagement := result.Get("membershipStates").Get(memberShipName).Get("servicemesh.dataPlaneManagement.state").String()
+							controlPLaneManagement := result.Get("membershipStates").Get(memberShipName).Get("servicemesh.controlPlaneManagement.state").String()
+							if dataPlaneManagement == "FAILED_PRECONDITION" || controlPLaneManagement == "FAILED_PRECONDITION" {
+								return false, fmt.Errorf("Service Mesh provisioning failed.")
+							} else if dataPlaneManagement == "PROVISIONING" || controlPLaneManagement == "PROVISIONING" {
```

Comment:
tought: we might actually want to know if the state went from `PROVISIONING` -> `ACTIVE` succesfully for both control plane and data plane, assuming errors can happen during provisioning

Reply:
FAILED_PRECONDITION should be the error status, shouldn't be?

Reply:
if it is on FAILED_PRECONDITION it was because an error occured, but does it have only 3 states? (ACTIVE, PROVISIONING, FAILED_PRECONDITION) 

I was not able to find the docs for this API, so I'm not sure. If we only have those 3 I think we don't need to verify if the state is in `ACTIVE` to stop retrying

Reply:
There are a few other possible status - however the only one we want before moving on is `ACTIVE`.